Optionally Enable Bugzilla Integration in ews-app

chgibb-apple opened the pull request:
#### 73d9ac867d56089d2fc651725095909056bdda68
<pre>
Optionally Enable Bugzilla Integration in ews-app
<a href="https://rdar.apple.com/153326078">rdar://153326078</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=294463">https://bugs.webkit.org/show_bug.cgi?id=294463</a>

Reviewed by NOBODY (OOPS!).

Optionally Enable Bugzilla Integration in ews-app

* Tools/CISupport/ews-app/ews/config.py:
* Tools/CISupport/ews-app/ews/fetcher.py:
(FetchLoop.run):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f793c604638be8a08b2f88880f18b15d8335338b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124446 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44129 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34861 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131290 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76457 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7856a20b-19a9-47e2-9faf-43b9f7fb56b5) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126323 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44869 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52724 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94678 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/62798 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b6248b57-a0ca-46dd-9ba6-7bbbff798af0) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127400 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35767 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111305 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75254 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/95f62766-7cf3-4755-9073-12ba2c7d0939) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34703 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29466 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74768 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105519 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29687 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/133955 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51343 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39173 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103153 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51744 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107522 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102942 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48304 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26561 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48288 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51209 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50626 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/53988 "Built successfully") | | | | 
| | [❌ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52303 "Hash f793c604 for PR 46723 does not build (failure)") | | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/ews-app/ews/config.py:
```diff
@@ -43,6 +43,8 @@
 BUILDBOT_TRY_USERNAME = util.load_password('BUILDBOT_TRY_USERNAME', default='sampleuser')
 BUILDBOT_TRY_PASSWORD = util.load_password('BUILDBOT_TRY_PASSWORD', default='samplepass')
 
+is_bugzilla_not_enabled = util.load_password('BUGZILLA_NOT_ENABLED') is not None
```
Nit: better to name it either: `BUGZILLA_DISABLED` or `BUGZILLA_ENABLED`. Current naming makes the statement like this hard to read: `if not is_bugzilla_not_enabled:`
Also please have the default value as the one which it is currently.

aj062 commented on Tools/CISupport/ews-app/ews/fetcher.py:
```diff
@@ -52,14 +52,15 @@ def run(self):
-            except Exception as e:
-                _log.error('Exception in BugzillaPatchFetcher: {}\n{}'.format(e, traceback.format_exc()))
-            time.sleep(self.interval)
+        if not is_bugzilla_not_enabled:
```
Can simply do an early return here, instead of nesting the code block.
e.g.:
```
if is_bugzilla_disabled:
    return
```

aj062 reviewed (commented):
PR title should say "Optionally disable" instead of "Optionally enable" bugzilla integration. Default should be enabled.

chgibb-apple committed:
Optionally Enable Bugzilla Integration in ews-app
rdar://153326078
https://bugs.webkit.org/show_bug.cgi?id=294463

Reviewed by NOBODY (OOPS!).

Optionally Enable Bugzilla Integration in ews-app

* Tools/CISupport/ews-app/ews/config.py:
* Tools/CISupport/ews-app/ews/fetcher.py:
(FetchLoop.run):
